src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,444 @@
+                handleFailure(format("Failed to find supply channel of key '%s'", supplyChannelKey), null);
+                return Optional.empty();
+            }
+        }
+        return Optional.of(draft);
+    }
+
+    /**
+     * Returns new {@link InventoryEntryDraft} containing same data as {@code draft} except for
+     * supply channel reference that is replaced by reference pointing to {@code supplyChannelId}.
+     *
+     * @param draft           draft where reference should be replaced
+     * @param supplyChannelId ID of supply channel existing in target project
+     * @return {@link InventoryEntryDraft} with supply channel reference pointing to {@code supplyChannelId}
+     *      and other data same as in {@code draft}
+     */
+    private InventoryEntryDraft ofEntryDraftPlusRefToSupplyChannel(@Nonnull final InventoryEntryDraft draft,
+                                                                   @Nonnull final String supplyChannelId) {
+        final Reference<Channel> supplyChannelRef = Channel.referenceOfId(supplyChannelId);
+        return InventoryEntryDraftBuilder.of(draft)
+            .supplyChannel(supplyChannelRef)
+            .build();
+    }
+
+    /**
+     * Method tries to create supply channel of given {@code supplyChannelKey} in CTP project.
+     * If operation succeed then {@code supplyChannelKeyToId} map is updated, otherwise
+     * error callback function is executed.
+     *
+     * @param supplyChannelKey key of supply channel that seems to not exists in a system
+     * @return {@link CompletionStage} instance that indicates method progress
+     */
+    private CompletionStage<Void> createMissingSupplyChannel(@Nonnull final String supplyChannelKey) {
+        return inventoryService.createSupplyChannel(supplyChannelKey)
+            .thenAccept(channel -> supplyChannelKeyToId.put(channel.getKey(), channel.getId()))
+            .exceptionally(ex -> {
+                handleFailure(format("Failed to create new supply channel of key '%s'", supplyChannelKey), ex);
+                return null;
+            });
+    }
+
+    /**
+     * Given a reason message as {@link String} and {@link Throwable} exception, this method calls the optional error
+     * callback specified in the {@code syncOptions}. <br/>
+     * <strong>NOTE:</strong> to the end of {@code message} following phrase will be joined:
+     * <pre> " in CTP project with key '_KEY_'"</pre>
+     * where _KEY_ is replaced by CTP project key, taken from CTP client configuration.
+     *
+     * @param message the reason of failure
+     * @param exception the exception that occurred, if any
+     */
+    private void handleFailure(@Nonnull final String message, @Nullable final Throwable exception) {
+        final String finalMessage = message + format(" in CTP project with key '%s.'",
+            syncOptions.getCtpClient().getClientConfig().getProjectKey());
+        syncOptions.applyErrorCallback(finalMessage, exception);
```

Comment:
I know that this is not within your changes but in general using those protected fields of
base class is not the best practice